WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift

```diff
@@ -120,15 +130,58 @@ private extension AddProductCoordinator {
                                        arrowDirections: .any)
     }
 
+    /// Creates & Fetches a template product.
+    /// If success: Navigates to the product.
+    /// If failure: Shows an error notice
+    ///
+    func createAndPresentTemplate(productType: BottomSheetProductType) {
+        guard let template = Self.templateType(from: productType) else {
+            DDLogError("⛔️ Product Type: \(productType) not supported as a template.")
+            return presentErrorNotice()
+        }
+
+        // Loading ViewController while the product is being created
+        let loadingTitle = NSLocalizedString("Creating Template Product...", comment: "Loading text while creating a product from a template")
+        let viewProperties = InProgressViewProperties(title: loadingTitle, message: "")
+        let inProgressViewController = InProgressViewController(viewProperties: viewProperties)
+
+        let action = ProductAction.createTemplateProduct(siteID: siteID, template: template) { result in
+            switch result {
+            case .success(let product):
+                // Dismiss the loader and present the product.
+                inProgressViewController.dismiss(animated: true)
```

Comment:
What do you think of moving this line to above the `switch` statement, to make sure the loader is dismissed in both success and failure cases? Right now, if the remote request fails the loader view is never dismissed:

<img src="https://user-images.githubusercontent.com/8658164/199249555-3dac5c4f-a1bc-4f6e-9e9a-6b96a273ef6f.png" width="300px">

Reply:
Nice catch! fixed in https://github.com/woocommerce/woocommerce-ios/pull/7992/commits/5269f3e15d66885aabe00bdabb2e1907f07ff22b


https://user-images.githubusercontent.com/562080/199252600-83b1ba90-46a7-4954-81a7-79525ba1d009.mov